interfaces/builtin/lxd_client.go

```diff
@@ -0,0 +1,79 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const lxdClientConnectedPlugAppArmor = `
+# Description: Can access commands and socket from the 'lxd' snap.
+/var/snap/lxd/common/lxd/unix.socket rw,
+`
+
+type LxdClientInterface struct{}
+
+func (iface *LxdClientInterface) Name() string {
+	return "lxd-client"
```

Comment:
This should be renamed as 'lxc' with the lxd snap adding 'slots: [ lxd ]' and clients using 'plugs: [ lxd ]'. See interfaces/builtin/docker.go for an example of what this might look like. We do this because as implemented, the 'lxc-client' interface is assuming there is something to connect to at /var/snap/lxd/common/lxd/unix.socket, but that is only there when the lxd snap is installed. Doing it as I suggest will allow the lxd snap to be auto-installed when installing a client that needs it (when that feature is implemented).